crates/holochain/src/conductor/conductor.rs

```diff
@@ -3298,6 +3308,102 @@ mod countersigning_impls {
                 .await?,
             )
         }
+
+        pub(crate) async fn get_countersigning_session_state(
+            &self,
+            cell_id: &CellId,
+        ) -> ConductorResult<Option<CountersigningSessionState>> {
+            let space = self.get_or_create_space(cell_id.dna_hash())?;
+            let countersigning_workspaces = space.countersigning_workspaces.lock();
+            match countersigning_workspaces.get(cell_id) {
+                None => Ok(None),
+                Some(workspace) => Ok(workspace.get_countersigning_session_state()),
+            }
+        }
+
+        pub(crate) async fn abandon_countersigning_session(
+            &self,
+            cell_id: &CellId,
+        ) -> ConductorResult<()> {
+            let space = self.get_or_create_space(cell_id.dna_hash())?;
+            let (countersigning_workspace, preflight_request) = self
+                .get_preflight_request_from_unknown_session(&space, cell_id)
+                .await?;
+            force_abandon_session(space.clone(), cell_id.agent_pubkey(), &preflight_request)
```

Comment:
Even if Holochain is waiting on network responses or in the middle of making database changes based on a decision, there aren't locks that are guaranteed to protect the entire process. I think the safer way to do this would be to not rely on just being in the unknown state. It would be better to mark the session and have the workflow abandon it instead. Then this couldn't happen in the middle of other state changes